source/syntax/CSTSerializer.cpp

```diff
@@ -0,0 +1,202 @@
+//------------------------------------------------------------------------------
+// CSTSerializer.cpp
+// Concrete Syntax Tree JSON serialization
+//
+// SPDX-FileCopyrightText: Michael Popoloski
+// SPDX-License-Identifier: MIT
+//------------------------------------------------------------------------------
+#include "slang/syntax/CSTSerializer.h"
+
+#include "slang/parsing/Token.h"
+#include "slang/syntax/AllSyntax.h"
+#include "slang/syntax/SyntaxTree.h"
+#include "slang/syntax/SyntaxVisitor.h"
+
+// This file contains CST JSON conversion functionality.
+
+namespace slang::syntax {
+
+// Define traits and utility functions for CSTJsonMode
+const std::array<CSTJsonMode, 4> CSTJsonMode_traits::values = {
+    CSTJsonMode::Full, CSTJsonMode::SimpleTrivia, CSTJsonMode::NoTrivia, CSTJsonMode::SimpleTokens};
+
+std::ostream& operator<<(std::ostream& os, CSTJsonMode mode) {
+    return os << toString(mode);
+}
+
+std::string_view toString(CSTJsonMode mode) {
+    switch (mode) {
+        case CSTJsonMode::Full:
+            return "Full";
+        case CSTJsonMode::SimpleTrivia:
+            return "SimpleTrivia";
+        case CSTJsonMode::NoTrivia:
+            return "NoTrivia";
+        case CSTJsonMode::SimpleTokens:
+            return "SimpleTokens";
+    }
+    SLANG_UNREACHABLE;
+}
+
+CSTSerializer::CSTSerializer(JsonWriter& writer, CSTJsonMode mode) : writer(writer), mode(mode) {
+}
+
+void CSTSerializer::serialize(const SyntaxTree& tree) {
+    writer.startObject();
+    writer.writeProperty("kind");
+    writer.writeValue(std::string_view("SyntaxTree"));
```

Comment:
Seems weird to always write "SyntaxTree" here... also why the explicit string_view constructor call?